sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ColumnDefaultValue.java

```diff
@@ -68,16 +64,20 @@ public Literal<?> getValue() {
   public boolean equals(Object o) {
     if (this == o) return true;
     if (!(o instanceof ColumnDefaultValue that)) return false;
-    return sql.equals(that.sql) && value.equals(that.value);
+    return Objects.equals(getSql(), that.getSql()) &&
+        Objects.equals(getExpression(), that.getExpression()) &&
+        value.equals(that.value);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(sql, value);
+    return Objects.hash(getSql(), getExpression(), value);
   }
 
   @Override
   public String toString() {
-    return "ColumnDefaultValue{sql='" + sql + "\', value=" + value + '}';
+    return String.format(
+        "ColumnDefaultValue{sql=%s, expression=%s, value=%s}",
```

Comment:
I removed `''` around sql. We had double quotes for strings. Int literals also showed up as string literals.